### ISSUE TITLE
mythicmobs and disabled world problem

### ISSUE THREAD

**Issue by hkkongou** [BG-Software-LLC/WildStacker#269]
### Minecraft's Version

This server is running Paper version git-Paper-780 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
You are 6 version(s) behind
Previous version: git-Purpur-1171 (MC: 1.16.5)

### Plugin's Version

WildStacker version 3.6.0-DEVb184
[17:15:44 INFO]: Stacking drops, entities, blocks and spawners into one object!
[17:15:44 INFO]: Website: https://bg-software.com/
[17:15:44 INFO]: Author: Ome_R

### Describe the bug

These error spamming in my console, i have checked from the WS discord, it is due to mythicmobs seems like.

![image](https://user-images.githubusercontent.com/61608872/129480952-f15ece37-3496-4caa-8c36-66436cb63779.png)

However, all of my mythicmobs are in a world called ``world_dungeon``, the problem for me is that i have set up disable-world in config so that ``world_dungeon`` should not stack, or affected by wildstack in terms of stacking item, mobs and spawners.

If the error is showing, then most likely wildstacker is interfering the disabled world, even the config is disabling. However, vanilla mobs are not stacking in the world, so the config setting should be working, but not fully working???

There is another evidence indicating that wildstacker is affecting the mythicmobs in the disabled world, mythicmobs are dropping bones and arrows which they should not, i have 3 server with synced settings, mythicmobs in other server does not drop bones and arrow, but in the server with wildstacker, mythicmobs mobs still drop bones. (This is hard to reproduce, or maybe a false evidence since i can only reproduce with advancedenchanment enchantments installed, however mobs still got affected in disabled-world is still a problem)

### To Reproduce

1. setup a world
2. spawn some mythicmobs
3. kill it

### Additional Information

config:
https://paste.gg/p/anonymous/3ea5fb58981644d79a9a450360652ff0

**Comment by OmerBenGera** [comment]
Does it only happen with MythicMobs?

**Comment by hkkongou** [comment]
> Does it only happen with MythicMobs, and only in disabled worlds?

no , mythicmobs in disabled world or non disabled world still dropping bones and arrow, those mobs are not configured to drop vanilla items like bones and arrows.

In a server without wildstacker, mythicmobs behave normal and no vanilla drops are dropped.

I don't have another custom mob plugin in my server, but here the biggest suspect is that wildstacker still affect mobs in disabled worlds.

**Comment by OmerBenGera** [comment]
Try adding MYTHIC_MOBS to your blacklist of entities, let me know if it fixes anything

**Comment by hkkongou** [comment]
> MYTHIC_MOBS

like this?

![image](https://user-images.githubusercontent.com/61608872/129529554-6294916a-eb81-4400-9c29-6b8549bd592b.png)

**Comment by OmerBenGera** [comment]
Don't add it to the blacklist as you have whitelist already enabled, therefore mobs of MythicMobs will not get stacked.
WildStacker should support MythicMobs in terms of drops and everything else, which is weird. I'll try to replicate the issue myself on my own server and get back to you.

**Comment by OmerBenGera** [comment]
I tested it, and I think it's a bug with MythicMobs. I completely removed WildStacker from the server, and custom mobs that don't have a custom drop table, but they use the drops list from the config, still drop vanilla drops. For example, using the default files of MythicMobs, SkeletalKnight drops vanilla drops alongside of the gold nuggets, however the SkeletonKing drops only custom drops.

I think you should contact MythicMobs regarding this, as it doesn't seem to be an issue with WildStacker.

**Comment by OmerBenGera** [comment]
Also, about the disabled worlds feature - it only prevents mobs from stacking in this world, the plugin however still handles drops etc, even tho it's not the case here, it's still good that you know that.

**Comment by hkkongou** [comment]
> Also, about the disabled worlds feature - it only prevents mobs from stacking in this world, the plugin however still handles drops etc, even tho it's not the case here, it's still good that you know that.

humm
then ,  is there anyway to fix the NAN health spam?

**Comment by OmerBenGera** [comment]
> 
> 
> > Also, about the disabled worlds feature - it only prevents mobs from stacking in this world, the plugin however still handles drops etc, even tho it's not the case here, it's still good that you know that.
> 
> humm
> then , is there anyway to fix the NAN health spam?

I couldn't replicate it, does it happen for you with default config files of MythicMobs?

**Comment by hkkongou** [comment]
> > > Also, about the disabled worlds feature - it only prevents mobs from stacking in this world, the plugin however still handles drops etc, even tho it's not the case here, it's still good that you know that.
> > 
> > 
> > humm
> > then , is there anyway to fix the NAN health spam?
> 
> I couldn't replicate it, does it happen for you with default config files of MythicMobs?
yep
just tried, something wrong when mobs got killed
![image](https://user-images.githubusercontent.com/61608872/129552495-aa8d4b60-7678-4689-be99-bb443e7cc8af.png)

**Comment by hkkongou** [comment]
seems like some problem with sweeping edge..
mobs die around the main mobs will trigger this error

**Comment by hkkongou** [comment]
just recorded a video to demonstrate the problem:
https://youtu.be/hLGlVS-U-3U

**Comment by OmerBenGera** [comment]
I can't seem to replicate the issue.
However, I checked the code of minecraft and this error can technically be ignored as of now.

I see you have McMMO - so first, try getting rid of it and test it again. If it still happens, mind sharing with me your MM files and which files did you try to kill? Also, let me know what weapon you used (and if it had enchantments, etc)?

**Comment by OmerBenGera** [comment]
Have you tested it?

**Comment by hkkongou** [comment]
> Have you tested it?

sorry for late reply and yes, this problem should be caused by mcmmo 
however i don't know what mcmmo setting is causing this, i have tried to change some mcmmo setting and suprisingly the error does not show anymore, but so far i forgot which one ..
humm

**Comment by OmerBenGera** [comment]
I will see what can be done regarding this.

**Comment by hkkongou** [comment]
update
after i update mcmmo and wipe my old settings and tune them as the old files, sweeping edge no longer trigger this NAN health problem, the NAN-HEALTH message have been reduced, but , it is still appearing, still finding why 🤔🤔
but most likely is due to mcmmo still

**Comment by OmerBenGera** [comment]
Any news regarding this?

**Comment by hkkongou** [comment]
> Any news regarding this?

still, is related to sweeping edge

**Comment by hkkongou** [comment]
try try these files....

[mcmmo+mtm.zip](https://github.com/BG-Software-LLC/WildStacker/files/7111793/mcmmo%2Bmtm.zip)

**Comment by hkkongou** [comment]
and this is my WS files...
thats all i got to reproduce, by using a sweeping edge sword to kill multiple mobs at the same time, nan health is spamming.

for mythicmobs files, they are default files
[WS.zip](https://github.com/BG-Software-LLC/WildStacker/files/7111795/WS.zip)

**Comment by OmerBenGera** [comment]
Does it happen when WS is uninstalled or not?

**Comment by hkkongou** [comment]
> Does it happen when WS is uninstalled or not?

no, this does not happen when WS is uninstalled, or WS is unloaded/disabled by plugman

i have done a test with video, with mythicmobs default mobs, mcmmo and WS, when WS loaded when server start, killing mobs with sweeping edge cause NAN-HEALTH, plugman disable WS does not have any warning 

files are attached above

https://youtu.be/a6-sp5113yU
